### PR TITLE
Bump spark3.version from 3.1.2 to 3.2.1 and update Spark SQL Extensions / Bump delta-core_2.12 from 1.0.0-nessie to 1.1.0-nessie

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: site/docs/tools/sql.md [hunk 1]
```diff
@@ -4,12 +4,12 @@ Spark SQL extensions provide an easy way to execute common Nessie commands via S
 ## How to use them
 
 In order to be able to use Nessie's custom Spark SQL extensions, one needs to configure
-`org.projectnessie:nessie-spark-extensions:{{ versions.java }}` along with `org.apache.iceberg:iceberg-spark3-runtime:{{ versions.iceberg }}`.
+`org.projectnessie:nessie-spark-extensions:{{ versions.java }}` along with `org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:{{ versions.iceberg }}`.
```
Comment (XN137) [inline site/docs/tools/sql.md:7]: wouldnt it make sense to have `{{ versions.iceberg-spark }}` as well?

File: pom.xml [hunk 2]
```diff
@@ -193,7 +193,7 @@
     <slf4j.version>1.7.36</slf4j.version>
     <soebes.itf.version>0.11.0</soebes.itf.version>
     <spark3.jackson.version>2.13.1</spark3.jackson.version>
-    <spark3.version>3.1.2</spark3.version>
+    <spark3.version>3.2.1</spark3.version>
```
Comment (XN137) [inline pom.xml:196]: just for my understanding:
why can we update this here and need to update all the docs to the spark version specific runtime... but we ourselves still use the "generic" spark runtime artifact from iceberg in our build
https://github.com/projectnessie/nessie/blob/c8b2becbb52e93381c320295c7b936fca6eeaf3d/pom.xml#L362-L365
?

File: site/docs/tools/sql.md (hunk 1 above)
Comment (snazy) [reply site/docs/tools/sql.md:7]: Makes sense

**Comment by nastra** [comment]
will come back to this PR once https://github.com/projectnessie/nessie/issues/3302 is done as it's required in order to be able to update to the latest Spark version

**Comment by rymurr** [comment]
I suppose if we put all the actual command logic into a class under a spark module then have a spark3.2 module extend it and `LeafCommand` and a spark 3.1 module extend it and `Command` we can build both and avoid duplication...any better ideas?

**Comment by nastra** [comment]
superseded by https://github.com/projectnessie/nessie/pull/3319

**Pull request closed without merging**